packages/dashboard/src/components/iot-dashboard/iot-dashboard.tsx

```diff
@@ -150,13 +168,11 @@ export class IotDashboard {
 
   onPaste() {
     const existingWidgetIds = this.getDashboardConfiguration().map(({ id }) => id);
-    this.setDashboardConfiguration(
-      paste({
-        dashboardConfiguration: this.getDashboardConfiguration(),
-        copyGroup: this.copyGroup,
-        numTimesCopyGroupHasBeenPasted: this.numTimesCopyGroupHasBeenPasted,
-      })
-    );
+
+    this.pasteWidgets({
+      copyGroup: this.copyGroup,
+      numTimesCopyGroupHasBeenPasted: this.numTimesCopyGroupHasBeenPasted,
```

Comment:
is it possible to have this logic contained within the reducer and handled internally rather than having the component be aware of this?